sample-service/src/main/java/uk/ac/ox/ndph/mts/sample_service/controller/MainController.java

```diff
@@ -44,7 +50,7 @@ public String getsecret() {
      *
      * @return a message as a string.
      */
-    @PreAuthorize("@authorisationService.authorise('stubPermission')") //NOSONAR
+    @PreAuthorize("@authenticationService.authenticate(env.getProperty(Managed_Identity)) or @authorisationService.authorise('stubPermission')") //NOSONAR
```

Comment:
I think authenticate is a bit confusing here. That is after we have passed the authentication and now authorising the request... 
Also `env.getProperty(Managed_Identity) `and the:
`private final static String Managed_Identity = "MANAGED_IDENTITY";
`
introduces authorisation logic and authorisation properties to the service. So I think it is better to move it to the AuthorisationService

Reply:
this if for a service to service call so we havent 'passed' authentication for a service call - I agree with you that the code introduces authentication logic into the service and Id prefer for it not to happen but to 'inject' the MSI env var into the authentication service we would need to have the authentication service as an app in azure so TF can configure it. We cant do that if teh authentication service is in a jar... I think...

Reply:
misunderstood the approach - Ive updated the code to remove the MSI being 'passed' to the auth service via the calling service.